packages/core/src/repositories/repository.ts

```diff
@@ -24,28 +21,6 @@ export default abstract class Repository<
     this.db = db
   }
 
-  /**
-   * This use $dynamic() query
-   * https://orm.drizzle.team/docs/dynamic-query-building
-   */
-  static async paginateQuery<T extends PgSelect>({
-    query,
-    page = 1,
-    pageSize = 20,
-  }: {
-    query: T
-  } & PaginationArgs): Promise<PaginatatedResult<T>> {
-    // @ts-ignore
-    query.config.fields = {
-      // @ts-ignore
-      ...query.config.fields,
-      __count: sql<number>`count(*) over()`,
-    }
-    const rows = await query.limit(pageSize).offset((page - 1) * pageSize)
-    const count = rows[0]?.__count ? Number(rows[0]?.__count) : 0
-    return [rows, count]
-  }
```

Comment:
Repositiries are not the right place to put pagination concerns. Now one doubt I have is with the generic `Repository.findAll()`. Now it executes the query. I think would be better to return the query and use the `paginateQuery` service to let it do the pagination work. I wont do this refactor here 

But what do you think?

Reply:
hmm i think it's weird ideally you want the repository.findall method to return the results not the query, any way we can integrate the pagination into the findall method?

Reply:
pagination is page aware in the sense that you have to pass a base URL to generate 
`< prev` and `next >` URLs. We can pass this prop to the `findAll` but it feels weird to me. For what I saw in Rails word pagination is aware of ORM but ORM is not aware of pagination. 

<img width="633" alt="image" src="https://github.com/user-attachments/assets/25b5e835-5666-4e49-b53c-1d0497328f74">

No big deal whatever you prefer

Reply:
hmm ya, well then yes i would remove the findall method from repositories all together and use the scope plus the new pagination method you created